src/main/java/com/fyle/bank/services/BankServiceImpl.java

```diff
@@ -0,0 +1,53 @@
+package com.fyle.bank.services;
+
+import java.util.List;
+import java.util.Optional;
+
+import javax.validation.constraints.NotBlank;
+
+import com.fyle.bank.models.BankBranches;
+import com.fyle.bank.models.Branch;
+import com.fyle.bank.repositories.BankBranchesRepository;
+import com.fyle.bank.repositories.BranchRepository;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BankServiceImpl implements BankService {
+
+    @Autowired
+    BranchRepository branchRepository;
+
+    @Autowired
+    BankBranchesRepository bankBranchesRepository;
+    
+    @Override
+    public Branch getBranch(@NotBlank String ifsc) {
+        Optional<Branch> branch = branchRepository.findById(ifsc);
+        if (!branch.isPresent())
+            throw new RuntimeException("No Branch associated with ifsc code");
+        return branch.get();
+    }
+
+    @Override
+    public Page<BankBranches> getPaginatedBranchesByCity(@NotBlank String name, @NotBlank String city, Pageable pageable) {
```

Comment:
why do we have this method when we are not using

Reply:
It's in use dude